src/lib.rs

```diff
@@ -499,7 +544,18 @@ impl<'h, 'b> Response<'h, 'b> {
         complete!(skip_empty_lines(&mut bytes));
         self.version = Some(complete!(parse_version(&mut bytes)));
         space!(bytes or Error::Version);
-        self.code = Some(complete!(parse_code(&mut bytes)));
+        self.code = match parse_code(&mut bytes) {
+            Ok(Status::Complete(code)) => Some(code),
+            Ok(Status::Partial) => return Ok(Status::Partial),
+            Err(Error::Status) if config.allow_multiple_spaces_in_response_status_delimiters => {
+                // The code failed to parse immediately following the first `SP` delimiter. If the
+                // option is enabled, try to consume additional spaces, and then try parsing the
+                // status code again.
+                complete!(skip_spaces(&mut bytes));
+                Some(complete!(parse_code(&mut bytes)))
+            },
+            Err(e) => return Err(e),
+        };
```

Comment:
@seanmonstar please check whether this reflects your not-actual-error idea described in Discord